test/Main.hs

```diff
@@ -0,0 +1,50 @@
+import Test.Tasty
+import Test.Tasty.QuickCheck hiding ((.&.))
+import Test.QuickCheck.Instances.Vector ()
+
+import Data.Bits
+import Data.Complex
+import qualified Data.Vector.Unboxed as V
+import Data.Vector.FFT
+
+expi :: Double -> Complex Double
+expi y = cos y :+ sin y
+
+dft :: V.Vector (Complex Double) -> V.Vector (Complex Double)
+dft v = V.generate len (\k -> V.sum $ V.imap (\n x -> x * expi (-2 * pi * fromIntegral k * fromIntegral n / fromIntegral len)) v)
+  where
+    len = V.length v
+
+idft :: V.Vector (Complex Double) -> V.Vector (Complex Double)
+idft v = V.generate len (\k -> (V.sum $ V.imap (\n x -> x * expi (2 * pi * fromIntegral k * fromIntegral n / fromIntegral len)) v) / fromIntegral len)
+  where
+    len = V.length v
+
+(~=~) :: V.Vector (Complex Double) -> V.Vector (Complex Double) -> Property
+v1 ~=~ v2 = counterexample (show v1 ++ (if res then " == " else " /= ") ++ show v2) res
+  where
+    epsilon = 1e-6
+    res = V.all (\(x, y) -> magnitude (x - y) < epsilon) (V.zip v1 v2)
```

Comment:
I'm a bit rusty on this stuff, but is the componentwise error a good indicator of reconstruction quality ? As compared to vector norms etc.

Reply:
Uhh, I'm not sure, since FFT is exact, so we're only measuring floating point errors. I think it shouldn't matter, since all norms on finite dimensional vector spaces are equivalent. [RustFFT](https://github.com/ejmahler/RustFFT/blob/master/tests/accuracy.rs#L27-L36) seems to measure the average distance instead.